docs/8-veba-troubleshooting.md

```diff
@@ -0,0 +1,277 @@
+vmware           tinywww-7fcfc6fb94-v7ncj               1/1     Running     1          6d4h
+vmware           vmware-event-router-5dd9c8f858-9g44h   1/1     Running     4          6d4h
+```
+
+First, we want to see if the event router is capturing events and forwarding them on to a function. 
+
+Use this command to follow the live Event Router log
+```
+kubectl logs -n vmware vmware-event-router-5dd9c8f858-9g44h --follow
+```
+
+For this sample troubleshooting, we have the sample hostmaintenance alarms function running. To see if the appliance is properly handling the event, we put a host into maintenance mode. 
+
+When we look at the log output, we see various entries regarding EnteredMaintenanceModeEvent, ending with the following:
+
+```
+
+[OpenFaaS] 2020/03/11 22:15:09 invoking function(s) on topic: EnteredMaintenanceModeEvent
+[OpenFaaS] 2020/03/11 22:15:09 successfully invoked function powercli-entermaint for topic EnteredMaintenanceModeEvent
+```
+
+This lets us know that the function was invoked. If we still don't see the expected result, we need to look at the function logs.
+
+Each OpenFaaS function will have its own pod running in the openfaas-fn namespace. We can examine the logs with the following command:
+
+```
+kubectl logs -n openfaas-fn powercli-entermaint-d84fd8d85-sjdgl
+```
+
+We don't need the --follow switch because we are just trying to look at recent logs, but --follow would work too.  
+Some other useful switches are --since and --tail. 
+
+This command will show you the last 5 minutes worth of logs:
+```
+kubectl logs -n openfaas-fn powercli-entermaint-d84fd8d85-sjdgl --since=5m
+```
+
+This command will show you the last 20 lines of logs
+```
+kubectl logs -n openfaas-fn powercli-entermaint-d84fd8d85-sjdgl --tail=20
+```
+
+Log output showing a succesful function invocation: 
+```
+Connecting to vCenter Server ...
+
+Disabling alarm actions on host: esx01.labad.int
+Disconnecting from vCenter Server ...
+
+2020/03/11 22:15:15 Duration: 6.085448 seconds
+```
+
+An alternative way to troubleshoot OpenFaaS logs is to use faas-cli
+
+This faas-cli command will show all available functions in the appliance. ```--tls-no-verify``` bypasses SSL certificate validation
+
+```
+faas-cli list --tls-no-verify
+```
+
+The command output is:
+```
+Function                        Invocations     Replicas
+powercli-entermaint             3               1
+```
+
+We can  look at the logs with this command:
+
+```
+faas-cli logs powercli-entermaint --tls-no-verify
+```
+
+The logs are the same: 
+
+```
+2020-03-11T22:15:15Z Connecting to vCenter Server ...
+2020-03-11T22:15:15Z
+2020-03-11T22:15:15Z Disabling alarm actions on host: esx01.labad.int
+2020-03-11T22:15:15Z Disconnecting from vCenter Server ...
+2020-03-11T22:15:15Z
+2020-03-11T22:15:15Z 2020/03/11 22:15:15 Duration: 6.085448 seconds
+```
+All of the same switches shown in the kubectl commands such as ```--tail``` and ```--since``` work with faas-cli
```

Comment:
Please allow for one empty line between (before/after) code blocks and text paragraphs.

Reply:
Fixed